numpy/core/src/umath/ufunc_object.c

```diff
@@ -5351,7 +5348,7 @@ ufunc_at(PyUFuncObject *ufunc, PyObject *args)
     }
 
 fail:
-
+    /* iter_buffer has already been deallocated, don't use NpyIter_Close */
     if (op1_array != (PyArrayObject*)op1) {
         PyArray_ResolveWritebackIfCopy(op1_array);
     }
```

Comment:
It's still ugly that we end up with an "unpaired" `PyArray_ResolveWritebackIfCopy` here, and I'm not sure this code is quite right.

First, I think there is another underlying API problem, this time in `PyArray_MapIterArrayCopyIfOverlap`. That API function creates WRITEBACK arrays but the API doesn't give any way to resolve them. I.e, there should probably be something like `PyArray_MapIterArrayClose`. 

But `PyArray_MapIterArrayCopyIfOverlap` is such a rarely used, and undocumented, part of the API that I don't really feel like trying to fix that right now. Probably we should have thought about it a few months ago when we introduced the WRITEBACKIFCOPY flag. Maybe someone should write docs for that function that make clear the caller is responsible for calling `PyArray_ResolveWritebackIfCopy`.

Leaving that aside, I also think the call to `PyArray_ResolveWritebackIfCopy` here should be added back in at line 5338 where it was before. I think we need to call it in both the fail and success branches. My understanding is that there are two ways we can get a WRITEBACK array in `ufunc_at`: First, `PyArray_MapIterArrayCopyIfOverlap` can make them, and it is apparently our responsibility to resolve them, and second `NpyIter_AdvancedNew` can create them, and those ones are managed by `NpyIter_Close`.

Reply:
Or maybe, the success case should have `ResolveWritebackIfCopy` and the fail case should have `DiscardWritebackIfCopy`.